src/stdlib/interpreterLazyTypeCheck.ts

```diff
@@ -0,0 +1,192 @@
+import { Value } from '../types'
+import {
+  isInterpreterThunk,
+  evaluateThunkOnce,
+  evaluateThunk
+} from '../interpreter/lazyInterpreter'
+import { is_pair, set_head, set_tail, head, tail } from './list'
+
+export function is_thunk(v: Value) {
+  if (isInterpreterThunk(v)) {
+    // Unwrap once to expose the argument.
+    const result = force_once(v)
+    return result !== null ? isInterpreterThunk(result) : false
+  } else {
+    return false
+  }
+}
+
```

Comment:
I'm not sure, but can the initial Type Checking methods can be used if you just always force the arguments ?

Reply:
T2 Lazy Evaluation team: Are you proposing a user-level function `is_thunk`?

Reply:
To @Ahmedjjj:
In our lazy interpreter, a thunk or value may possibly be 'thunked' multiple times because function arguments are delayed. To better explain, here is an example:
`is_thunk(a);`
Assuming that `a` contains a thunk, then by delaying the arguments of the function, `a` will be 'thunked' once again.
In the body of `is_thunk(v: Value) { .. }`, it will first check if `v` is a thunk (returns `true` since arguments are always delayed except for some built-in functions that remains strict).
Then it will force the value of `v` **once** to expose the actual argument which is the identifier named `a`. Since we know that `a` contains a thunk (whether its already evaluated or not is not considered), then it will return `true`.

To @martin-henz:
It was originally created to help us with minor debugging in the interpreter. I'm not sure how useful it is for the users, but I'm more inclined to the idea that it should be taken down as the concept of thunks need not be transparent to the users.

Reply:
ok